Add custom hit-shapes and TargetableOffsets to RA structures

reaperrr opened the pull request:
RA equivalent to #13196.

reaperrr committed:
Align RA base building hitshapes with their sprites

reaperrr committed:
Set custom hit shapes for civilian buildings

reaperrr commented:
Currently contains the code changes from #13196, so one of them will need a rebase (probably this one).

pchote commented on mods/ra/rules/defaults.yaml:
```diff
@@ -491,6 +491,8 @@
 	Inherits@2: ^IronCurtainable
 	Inherits@3: ^SpriteActor
+	BodyOrientation:
+		CameraPitch: 255
```
This changes the shape of armament muzzles as they rotate with facing.  This was previously tuned to look correct, so I am assuming that this will break that.  Unless this was specifically to fix that?

pchote replied:
This regresses at least the turret firing offsets.

reaperrr replied:
I forgot about the defenses when setting this, but it's easy to fix as all defenses use footprint offsets and therefore don't really need the 'top-down' pitch.

pchote commented:
Some of the targetable postions (e.g. powr, both barracks, airfields) are floating outside the building artwork.  Would you mind doing some tweaks to put them inside?

reaperrr commented:
> Some of the targetable postions (e.g. powr, both barracks, airfields) are floating outside the building artwork. Would you mind doing some tweaks to put them inside?

Well, it's a bit tedious, but I guess better to do it now than needing another follow-up PR.

reaperrr committed:
Add building hitshape defaults to RA

And do some cleanup.
Intentionally omitted uncommon shapes that only appear once or twice.

reaperrr commented:
Rebased and updated. Travis didn't trigger, but it passed `make test` locally and no longer contains code, so it shouldn't matter.

pchote commented:
Needs a rebase

reaperrr committed:
Increase ZOffset of RA explosions

To make sure they play on top of actors.

reaperrr commented:
Rebased.

MustaphaTR commented:
Tesla weaponary still target the center of actor, not closet target. I tested with Shock Trooper and Tesla Coil.

![adsiz](https://user-images.githubusercontent.com/7933210/26884080-344a62c8-4b9f-11e7-8bd2-dc61d700c1f7.png)

Sov Tech Center, Radar Dome, Techology Center and Tech Oil Derrick also looks like they want some adjusments of Target Positions according to artwork.

![oild](https://user-images.githubusercontent.com/7933210/26884139-550c1cae-4b9f-11e7-8cd5-633c721e6258.png)

![techno](https://user-images.githubusercontent.com/7933210/26884756-de3897ea-4ba0-11e7-8d2e-ed58eb514453.png)

reaperrr commented:
> Tesla weaponary still target the center of actor, not closest target. I tested with Shock Trooper and Tesla Coil.

Fixed in #13475 to avoid cluttering this PR.

reaperrr commented:
Updated:

![new-t-pos](https://user-images.githubusercontent.com/2857877/26894133-ce2e391e-4bbd-11e7-81d7-39689711f8a9.png)

MustaphaTR reviewed (approved):
👍

pchote commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e8198bdb0159f1cd261339a45bbfbfec884c69ef

pchote commented:
In one of the followup balance PRs I think we should add `TargetCenterPosition: true` to artillery and cruisers.  It looks rather odd for them to try and aim at the corner of a building when their inaccuraccy is of the order of the building size.

reaperrr commented:
> In one of the followup balance PRs I think we should add `TargetCenterPosition: true` to artillery and cruisers.

That will need another partial Attack*/Armament/projectiles rewrite, just sayin'.
